### PR TITLE
t: Avoid tampering with git checkout

### PR DESCRIPTION
**Pull request by perlpunk**
I was running this test locally and it failed every second time, because it created some directory that wasn't cleaned up.

Now it copies everything it needs to a tempdir and leaves the source dir alone.

Related issue: https://progress.opensuse.org/issues/164883

### CODE REVIEW

**Review by okurz — APPROVED** [review]

File: t/14-grutasks.t [hunk 1]
```diff
@@ -33,15 +33,23 @@ use Mojo::Message::Response;
...
 use Time::Seconds;
 
 plan skip_all => 'set HEAVY=1 to execute (takes longer)' unless $ENV{HEAVY};
 
+# Avoid tampering with git checkout
+my $workdir = tempdir("$FindBin::Script-XXXX", TMPDIR => 1);
+my $guard = scope_guard sub { chdir $FindBin::Bin };
+chdir $workdir;
+mkdir 't';
+dircopy "$FindBin::Bin/$_", "$workdir/t/$_" for qw(data testresults fixtures);
```
Comment (Martchus) [inline t/14-grutasks.t:47]:
```suggestion
dircopy "$FindBin::Bin/$_", "$workdir/t/$_" or BAIL_OUT($!) for qw(data testresults fixtures);
```

File: t/14-grutasks.t [hunk 2]
```diff
@@ -241,8 +249,8 @@ is(
 $assets->update({size => 26 * $gib});
 run_gru_job($t->app, 'limit_assets');
 
-is(scalar @{mock_removed()}, 1, "one asset should have been 'removed' at size 26GiB");
-is(scalar @{mock_deleted()}, 1, "one asset should have been 'deleted' at size 26GiB");
+is(scalar @{mock_removed()}, 1, "assets should have been 'removed' at size 26GiB");
+is(scalar @{mock_deleted()}, 1, "assets should have been 'deleted' at size 26GiB");
```
Comment (Martchus) [inline t/14-grutasks.t:253]: I don't really care but would personally prefer the singular. We could also simplify the phrasing:
```suggestion
is(scalar @{mock_removed()}, 1, "asset is 'removed' at size 26GiB");
is(scalar @{mock_deleted()}, 1, "asset is 'deleted' at size 26GiB");
```

File: t/14-grutasks.t [hunk 3]
```diff
@@ -275,8 +283,8 @@ reset_mocked_asset_deletions;
 $assets->update({size => 34 * $gib});
 run_gru_job($t->app, 'limit_assets');
 
-is(scalar @{mock_removed()}, 1, "two assets should have been 'removed' at size 34GiB");
-is(scalar @{mock_deleted()}, 1, "two assets should have been 'deleted' at size 34GiB");
+is(scalar @{mock_removed()}, 1, "assets should have been 'removed' at size 34GiB");
+is(scalar @{mock_deleted()}, 1, "assets should have been 'deleted' at size 34GiB");
```
Comment (Martchus) [inline t/14-grutasks.t:287]:
```suggestion
is(scalar @{mock_removed()}, 1, "asset is 'removed' at size 34GiB");
is(scalar @{mock_deleted()}, 1, "asset is 'deleted' at size 34GiB");
```

File: t/14-grutasks.t [hunk 4]
```diff
@@ -852,5 +860,6 @@ done_testing();
 # break subsequent tests; can happen if a subtest creates a task but
 # does not execute it, or we crash partway through a subtest...
 END {
+    chdir $FindBin::Bin;
```
Comment (Martchus) [inline t/14-grutasks.t:863]: Do we need this with the scope guard?

Comment (perlpunk) [reply t/14-grutasks.t]: I don't know. I had some problems in the failing case, but I removed it now.

**Review by Martchus — APPROVED** [review]

**Pull request merged**